packages/directory/addon/components/dir-new-button.js

```diff
@@ -31,15 +32,14 @@ export default Component.extend({
   },
 
   /**
-   * @override
-   * @method init
+   * @property {Array} fileTypeNames - Names of file types in directory
    */
-  init() {
-    this._super(...arguments);
+  fileTypeNames: computed(function() {
```

Comment:
How about adding these fetch types and typeNames to the enum? I expect we will be reusing these methods a bit.